### PR TITLE
Interceptors 2.1.0 Release Review

### PR DESCRIPTION
**Pull request by starksm64**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of jakarta-wombat-spec-x.y.pdf
- [x] Specification HTML in the form of jakarta-wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable. N/A
- [x] For a [Release Review](https://www.eclipse.org/projects/handbook/#release-review), a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification. The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
   - [x] Updated release record
   - [x] Generated IP Log
   - [x] Email to PMC
   - [x] Start release review by emailing EMO 
    https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/176
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/interceptor/jakarta.interceptor-api/2.1.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      [CDI 4.0 TCK](https://download.eclipse.org/ee4j/cdi/4.0/cdi-tck-4.0.0-dist.zip)
- [x] The URL of the compatibility certification request issue:
      [Issue 96](https://github.com/eclipse-ee4j/interceptor-api/issues/96)
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

Note: If any item does not apply, check it and mark N/A below it.

Done.

### CODE REVIEW

File: interceptors/2.1/_index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Ballots
+
+## Plan Review
+
+The Specification Committee Ballot TBD.
+
+| Representative                                 | Representative for: | Vote |
+|------------------------------------------------|---------------------|------|
+| Kenji Kazumura                                 | Fujitsu             | +1   |
+| Dan Bandera, Kevin Sutter                      | IBM                 | +1   |
```
Comment (Emily-Jiang) [inline interceptors/2.1/_index.md]: The name needs to be updated

File: interceptors/2.1/_index.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| Ed Bratt, Dmitry Kornilov                      | Oracle              | +1   |
+| Andrew Pielage, Matt Gill                      | Payara              | +1   |
+| David Blevins, Jean-Louis Monteiro             | Tomitribe           | +1   |
+| Ivar Grimstad                                  | EE4J PMC            | +1   |
+| Marcelo Ancelmo, Martijn Verburg               | Participant Members | +1   |
+| Werner Keil                                    | Committer Members   | +1   |
+| Scott (Congquan) Wang                          | Enterprise Members  | +1   |
```
Comment (Emily-Jiang) [inline interceptors/2.1/_index.md]: The total number should be 10. One is missing.

File: interceptors/2.1/_index.md (hunk 1 above)
Comment (Emily-Jiang) [reply interceptors/2.1/_index.md]: ah, I saw it was fixed.

File: interceptors/2.1/_index.md (hunk 2 above)
Comment (Emily-Jiang) [reply interceptors/2.1/_index.md]: The later one commit fixed this.

**Comment by Emily-Jiang** [comment]
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form jakarta-{spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form jakarta-{spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to project release plan of the form https://projects.eclipse.org/projects/ee4j.{spec}/releases/x.y/plan
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip (The folder path is required, the file name pattern is preferred.)
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request

10. If a Release Review is required, the specification project team contacts the EMO to initiate the review by sending an email to emo@eclipse.org.
    (A Release Review is not required if the current release is a Service Release based on a previously successful Major or Minor
    release as indicated by a release record on the project's Releases page, e.g., the [Jakarta Servlet releases page](https://projects.eclipse.org/projects/ee4j.servlet/reviews).)
    - [ ] An issue will be created by the EMO to track the release review.
    - [ ] The specification project team requests approval for the release from the PMC by sending an email to ee4j-pmc@eclipse.org.
    - [ ] The specification project team then delivers an IP Log to the IP Team for their review as described in https://www.eclipse.org/projects/handbook/#pmi-commands-iplog.

11. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by Emily-Jiang** [comment]
@starksm64 I have the following comments regarding this PR:

1. The _index.md contains a few broken links to the spec docs etc. I will then check the licenses etc against the spec docs.
2. Since this spec has no tcks. I am wondering how to clarify about the compatible impelementation. You put weld there but there is no proof regarding the tck results. You might need to explain cdi tcks for interceptors passed.

**Comment by starksm64** [comment]
The links are working for me as seen at the preview page:
https://deploy-preview-450--jakartaee-specifications.netlify.app/specifications/interceptors/2.1/

**Comment by starksm64** [comment]
In the past the ejb tests from the platform TCK was used. I don't see why the CDI TCK's use of interceptors could also not be used. I'll create a CI against the interceptors project based on the CDI TCK and Weld this week.

**Comment by Emily-Jiang** [comment]
@starksm64 The spec pdf and html look good. The javadoc does not have the following:

- [ ] Footer contains Eclipse copyright and link to license
- [ ] ESFL license is included, usually as doc-files/speclicense.html

**Comment by starksm64** [comment]
The footer and license link are there, but when you first browse https://deploy-preview-450--jakartaee-specifications.netlify.app/specifications/interceptors/2.1/apidocs/jakarta/interceptor/package-summary.html there is a stupid overlay from netlify that is obscuring it:
spec-license-screenshot-overlay

Collapse that by clicking on the two arrows on the far right and you see:
spec-license-screenshot

**Comment by starksm64** [comment]
This should be ready for final review

**Comment by Emily-Jiang** [comment]
@starksm64 

> The footer and license link are there, but when you first browse https://deploy-preview-450--jakartaee-specifications.netlify.app/specifications/interceptors/2.1/apidocs/jakarta/interceptor/package-summary.html there is a stupid overlay from netlify that is obscuring it: spec-license-screenshot-overlay
> 
> Collapse that by clicking on the two arrows on the far right and you see: spec-license-screenshot

Thanks @starksm64 ! It was well hidden! got it now.

**Comment by Emily-Jiang** [comment]
@starksm64 can you add a link to final TCK download zip file of the form [https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip](https://download.eclipse.org/jakartaee/%7Bspec%7D/x.y/*%7Bspec%7D-tck-x.y.z.zip) (The folder path is required, the file name pattern is preferred.)

File: interceptors/2.1/_index.md [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
-The 2.1 release updates dependencies for EE 10 and add a JPMS module-info descriptor.
+The 2.1 release updates dependencies for EE 10 and adds a JPMS module-info descriptor.
 
 * [Jakarta Interceptors 2.1 Release Record](https://projects.eclipse.org/projects/ee4j.interceptors/releases/2.1)
-* [Jakarta Interceptors 2.1 Specification Document](./interceptors-spec-2.1.pdf) (PDF)
-* [Jakarta Interceptors 2.1 Specification Document](./interceptors-spec-2.1.html) (HTML)
+* [Jakarta Interceptors 2.1 Specification Document](./jakarta-interceptors-spec-2.1.pdf) (PDF)
+* [Jakarta Interceptors 2.1 Specification Document](./jakarta-interceptors-spec-2.1.html) (HTML)
 * [Jakarta Interceptors 2.1 Javadoc](./apidocs)
 * [Jakarta EE Platform 10 TCK](TBD) ([sig](TBD),[sha](TBD),[pub](TBD))
```
Comment (Emily-Jiang) [inline interceptors/2.1/_index.md:16]: I think this line should state: no TCK delivery for this specification instead of providing an invalid link - TBD

**Comment by Emily-Jiang** [comment]
@starksm64 the [link](https://deploy-preview-450--jakartaee-specifications.netlify.app/) is no longer valid

**Comment by starksm64** [comment]
I don't know how to trigger an update to the generated preview if a push does not do it.

**Comment by Emily-Jiang** [comment]
The link is back online. I just added a copyright comment to correct the end year.

**Comment by Emily-Jiang** [comment]
@starksm64 the javadoc footer license still says Copyright © 2018,2020 Eclipse Foundation. Can you update that to say 2018,2022?

**Comment by starksm64** [comment]
Copyright dates should be correct now

**Comment by starksm64** [comment]
Ok, set javadoc start date to 2018 since that seems most correct.

File: interceptors/2.1/apidocs/jakarta/interceptor/package-tree.html [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+<ul class="navList" title="Navigation">
+<li><a href="package-summary.html">Package</a></li>
+<li>Class</li>
+<li>Use</li>
+<li class="navBarCell1Rev">Tree</li>
+<li><a href="../../deprecated-list.html">Deprecated</a></li>
+<li><a href="../../index-all.html">Index</a></li>
+<li><a href="../../help-doc.html">Help</a></li>
+</ul>
+<div class="aboutLanguage"><br>Jakarta Interceptors API v2.1.0-SNAPSHOT</div>
```
Comment (kazumura) [inline interceptors/2.1/apidocs/jakarta/interceptor/package-tree.html:56]: Should not be include "-SNAPSHOT".

**Review by kazumura — CHANGES_REQUESTED** [review]

File: interceptors/2.1/apidocs/jakarta/interceptor/package-tree.html (hunk 4 above)
Comment (Emily-Jiang) [reply interceptors/2.1/apidocs/jakarta/interceptor/package-tree.html:56]: @starksm64 can you make the above change request from @kazumura ? Thank you @kazumura for spotting this!

File: interceptors/2.1/_index.md [hunk 5]
```diff
@@ -1,26 +1,29 @@
 ---
 title: "Jakarta Interceptors 2.1"
-date: 2022-01-12
+date: 2022-01-26
 summary: "Release for Jakarta EE 10"
 ---
-Jakarta Interceptors defines a means of interposing on business method invocations and specific
+Jakarta Interceptors dpwdefines a means of interposing on business method invocations and specific
```
Comment (Pandrex247) [inline interceptors/2.1/_index.md:6]: Typo
```suggestion
Jakarta Interceptors defines a means of interposing on business method invocations and specific
```

**Review by kazumura — APPROVED** [review]

**Comment by Emily-Jiang** [comment]
- [ ] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<include url of Issue here>

**Pull request merged**